src/core/mac/mac.cpp

```diff
@@ -870,6 +882,8 @@ extern "C" void otPlatRadioTransmitDone(otInstance *aInstance, otRadioFrame *aFr
 {
     otLogFuncEntryMsg("%!otError!, aRxPending=%u", aError, aRxPending ? 1 : 0);
 
+    aInstance->mThreadNetif.GetMac().SetFramePending(aRxPending);
```

Comment:
We want to stay awake if we have more frames to send. So here I think we should check the "frame pending" on the frame we sent. `aRxPending` is for the ack packet.